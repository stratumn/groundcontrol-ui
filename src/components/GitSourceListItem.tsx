// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React from "react";
import { Button, List } from "semantic-ui-react";

import { createFragmentContainer } from "react-relay";

import { GitSourceListItem_item } from "./__generated__/GitSourceListItem_item.graphql";

export interface IProps {
  item: GitSourceListItem_item;
  onDelete: (values: IProps) => any;
}

export function GitSourceListItem(props: IProps) {
  const{ item: { repository, reference }, onDelete } = props;
  const handleDelete = () => onDelete({ ...props });

  return (
    <List.Item>
      <List.Content>
        <Button
          floated="right"
          icon="delete"
          color="pink"
          size="small"
          onClick={handleDelete}
        />
        <List.Header>{repository}@{reference}</List.Header>
        <List.Description>Git Repository</List.Description>
      </List.Content>
    </List.Item>
  );
}

export default createFragmentContainer(GitSourceListItem, graphql`
  fragment GitSourceListItem_item on GitSource {
    id
    repository
    reference
  }`,
);
