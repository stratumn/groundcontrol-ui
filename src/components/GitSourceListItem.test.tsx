// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";
import { mocked } from "ts-jest/utils";

import { mockQueryPropAttrs } from "../testing/relay";

import { GitSourceListItem } from "./GitSourceListItem";

const props = {
  item: {
    ...mockQueryPropAttrs(),
    id: "id",
    reference: "reference",
    repository: "repository",
  },
  onDelete: jest.fn(),
};

beforeEach(() => {
  mocked(props.onDelete).mockClear();
});

describe("<GitSourceListItem />", () => {

  it("renders correctly", () => {
    const wrapper = shallow(<GitSourceListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("triggers onDelete when the delete button is clicked", () => {
    const wrapper = shallow(<GitSourceListItem {...props} />);
    wrapper.find("Button[icon='delete']").simulate("click");
    expect(props.onDelete).toBeCalledTimes(1);
    expect(props.onDelete).toBeCalledWith(props);
  });

});
