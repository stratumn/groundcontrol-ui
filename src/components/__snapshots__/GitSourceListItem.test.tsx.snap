// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitSourceListItem /> renders correctly 1`] = `
<ListItem>
  <ListContent>
    <Button
      as="button"
      color="pink"
      floated="right"
      icon="delete"
      onClick={[Function]}
      size="small"
    />
    <ListHeader>
      repository
      @
      reference
    </ListHeader>
    <ListDescription>
      Git Repository
    </ListDescription>
  </ListContent>
</ListItem>
`;
